js/helpers/src/index.ts

```diff
@@ -64,6 +65,49 @@ export async function runAllSolidityTests(
   });
 }
 
+/**
+ * Run all the given solidity tests and returns the whole results after finishing, including the gas report.
+ */
+export async function runAllSolidityTestsWithGasReport(
```

Comment:
Suggestion: Shouldn't we modify the original function? That one should also wait for the `Promise` to resolve, rather than waiting until `resultsFromCallback.length === testSuites.length`, since we now have a promise-based implementation rather than a function that kicks off a background processes and is informed using callback-driven results.